src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2261,7 +2261,12 @@ public static function registerRateLimiter(ContainerBuilder $container, string $
 
         $storageId = $limiterConfig['storage_service'] ?? null;
         if (null === $storageId) {
-            $container->register($storageId = 'limiter.storage.'.$name, CacheStorage::class)->addArgument(new Reference($limiterConfig['cache_pool']));
+            $cachePoolId = $limiterConfig['cache_pool'];
+            if ('cache.app' === $cachePoolId) {
+                $container->setDefinition($cachePoolId = 'cache.rate_limiter.'.$name, new ChildDefinition('cache.app'))->addTag('cache.pool');
+            }
```

Comment:
I'm not completely sure about this:

* Should we always create a namespaced cache pool? (also when a user overrides the `cache_pool` option with a custom one?)
* Should we create 1 rate limiter cache pool, or 1 for each rate limiter (like done here)?

Reply:
I think it'd be better to have one single `cache.rate_limiter` pool, that'd be defined in service configuration (and removed in the extension when not used by any limiter maybe?)

This might require prefixing keys so that different limiters don't conflict. Is that the case already? Otherwise, this is a good reason for using several cache pool. If using several cache pools, this should be done by having an abstract `cache.rate_limiter` pool service, and having ChildDefinition for each limiter (tagged as cache.pool)

Reply:
Thanks for the guidelines. I've updated the PR to use a single cache pool (cache keys are prefixed by the limiter ID - which is unique).

I think (hope) making the service private should be enough to remove it when it's not used?